### DISCUSSION TITLE
如何正确的卸载 ServiceWorker

### DISCUSSION THREAD

**Discussion by javahuang** [umijs/umi#7516] [Show and tell]
最近遇到了一个问题，某些用户的电脑打开的系统一直是旧的，不能更新。
原因是我们用的 umi2，默认开启了 pwa（太坑了），导致所有的静态资源文件（html、css、js）都是 200 OK (from service worker)，导致即使断网网站也能正常打开。
搜遍了 umi 的相关 issue，也没能解决，看到 umi 作者竟然也让手动清浏览器缓存，-_-||  

首先，即使调整了pwa 的配置，在部署路径的根目录下面也不给 service-worker.js 文件直接删掉。因为其它文件都是从 service worker 直接加载的，只有 `service-worker.js` 文件才会从后台请求。

下面介绍，如何正确的卸载。

```js
// 将 service-worker.js 内容替换为下面代码就可以了
self.registration.unregister();
```

至于其他答案回复的，如 `serviceWorker.getRegistrations()` 或者 `serviceWorker.unregister();`，我这边试都不行，因为 serviceWorker 根本没有这个方法...

**Comment by nianyi778** [comment]
https://blog.csdn.net/qq_43265673/article/details/118676527

**Reply by nianyi778** [reply]
这个能解决
